src/aggregator/aggregator/entry_test.go

```diff
@@ -1121,6 +1121,62 @@ func TestAddUntimed_ResendEnabledMigrationRace(t *testing.T) {
 	require.True(t, ok)
 }
 
+func TestAddUntimed_ResendEnabledMigrationRaceWithFlusher(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	e, _, _ := testEntry(ctrl, testEntryOptions{})
+	metadatas := metadata.StagedMetadatas{
+		{
+			Metadata: metadata.Metadata{
+				Pipelines: []metadata.PipelineMetadata{
+					{
+						AggregationID: aggregation.MustCompressTypes(aggregation.Sum),
+						ResendEnabled: false,
+						StoragePolicies: policy.StoragePolicies{
+							testStoragePolicy,
+						},
+					},
+				},
+			},
+		},
+	}
+	resolution := testStoragePolicy.Resolution().Window
+	mu := testGauge
+
+	// add value with resendEnable=false
+	require.NoError(t, e.addUntimed(mu, metadatas))
+	require.Len(t, e.aggregations, 1)
+	require.False(t, e.aggregations[0].resendEnabled)
+	elem := e.aggregations[0].elem.Value.(*GaugeElem)
+	vals := elem.values
+	require.Len(t, vals, 1)
+	t1 := xtime.ToUnixNano(e.nowFn().Truncate(resolution))
+	_, ok := vals[t1]
+	require.True(t, ok)
+
+	t2 := t1.Add(resolution)
+	// partially consume the aggregation
+	elem.dirtyToConsumeWithLock(int64(t2), resolution, isStandardMetricEarlierThan)
+
+	// add value with resendEnabled=true that targets the aggregation being flushed
+	metadatas[0].Metadata.Pipelines[0].ResendEnabled = true
+	mu.ClientTimeNanos = t1
+	require.NoError(t, e.addUntimed(mu, metadatas))
+
+	// continue consuming the aggregation
+	elem.expireValuesWithLock(int64(t2), isStandardMetricEarlierThan, flushMetrics{})
+
+	// target the aggregation being flushed again...it should still be open since it migrated to resendEnabled before
+	// closing.
+	require.NoError(t, e.addUntimed(mu, metadatas))
```

Comment:
this test fails without the code fix and produces the expected "aggregation closed" error